app/controllers/mixins/ems_common_angular.rb

```diff
@@ -75,7 +75,7 @@ def update_ems_button_validate(verify_ems = nil)
         add_flash(_("Credential validation was not successful: %{details}") % {:details => details}, :error)
       end
 
-      render_flash
+      render :json => {:message => @flash_array.last(1)[0][:message], :level => @flash_array.last(1)[0][:level]}
```

Comment:
What happens when there's more than one flash message?

Also, `.last(1)[0]` is `.last`. ;)